Add default Code of Conduct, Security policy for organization and internal repository linting files

d33bs opened the pull request:
This PR adds default Code of Conduct (`CODE_OF_CONDUCT.md`) and Security (`SECURITY.md`) policy files to protect the artifacts and people involved with the CU-DBMI organization. By including these files in this repository they will be included by default with every repository in the organization unless they are overridden within specific repositories.

In addition, I've added pre-commit checks along with a GitHub Actions workflow which runs the same to help ensure the work is kept at a minimum standard moving forward.

the-mayer reviewed (approved):
Looks good -- nothing to add at this time

falquaddoomi reviewed (approved):
This all looks good, but it looks like it also includes some pre-commit checks. While I think it's a great idea to pull this in at the org level, I wonder if you should update the description of the PR to say it includes those workflows, just for reference later on.

vincerubinetti commented:
Just to clarify, the references to Python in the workflows and pre-commit config are just because you're running the pre-commits with Python? It's not checking anything for the Python language specifically? Because of course we have several repos that have non-Python code in them.

Assuming I'm understanding correctly, writing and pre-commit checks look good to me.

d33bs commented:
Thanks @the-mayer, @falquaddoomi, and @vincerubinetti for the reviews here! 

> While I think it's a great idea to pull this in at the org level, I wonder if you should update the description of the PR to say it includes those workflows, just for reference later on.

Thanks, good point, I've updated the PR title and description to help make this clear.

> Just to clarify, the references to Python in the workflows and pre-commit config are just because you're running the pre-commits with Python? It's not checking anything for the Python language specifically? Because of course we have several repos that have non-Python code in them.

The `pre-commit-config.yaml` file includes a Python version reference which becomes the basis by which is implements checks through Python (but not for Python code it might check). It shouldn't have an impact outside of this repository.  My understanding is that while the `.github` repo sets default files for the org, the `.github/workflows` directory under that same repo still only applies to just this repo. Thanks just the same for this check however, as relatedly, it might have applied the `dependabot.yml` config to all repositories (this is a bit unclear so I removed for safety).
